superduperdb/base/datalayer.py

```diff
@@ -908,6 +907,29 @@ def _add(
         jobs.extend(these_jobs)
         return jobs
 
+    def _change_component_reference_prefix(self, serialized):
+        """Replace '?' to '&' in the serialized object."""
+        references = set()
+        for reference in list(serialized['_leaves'].keys()):
+            if isinstance(serialized['_leaves'][reference], Component):
+                serialized['_leaves'].pop(reference)
+                references.add(reference)
+
+        # Only replace component references
+        if not references:
+            return
+
+        def replace_function(value):
+            # Change value if it is a string and starts with '?'
+            # and the value is in references
+            # ?:xxx: -> &:xxx:
+            if isinstance(value, str) and value[1:] in references:
```

Comment:
When inserting a serialized item into the database, since the children have already been inserted, we change the reference from `?:component:/xxxxx/{uuid}` to `&:component:/xxxxx/{uuid}`. 

This means that we no longer need to search for children in _leaves, as they are already stored in the database. 

When loading, we simply call `db.load(uuid=uuid)` using the reference `&:component:/xxxxx/{uuid}`. 

Thus, it is necessary to delete the corresponding data from `_leaves`, as retaining redundant information is not required.

If not inserting into the database, we would use `?:component:/xxxxx/{uuid}` to search for information in _leaves to load it.

Saving blobs and files follows the same logic; once the data has been saved in the database through the insertion process, there is no need to keep the actual data in the dictionary, just maintaining the reference is sufficient.